internal/rpcserver/rpcserver.go

```diff
@@ -4669,14 +4669,22 @@ func handleSendRawTransaction(_ context.Context, s *Server, cmd interface{}) (in
 		// deserialization error code (to match bitcoind behavior).
 		var rErr mempool.RuleError
 		if errors.As(err, &rErr) {
-			err = fmt.Errorf("rejected transaction %v: %w", tx.Hash(),
+			hash := tx.Hash()
+			err = fmt.Errorf("rejected transaction %v: %w", hash,
 				err)
 			log.Debugf("%v", err)
 			if errors.Is(rErr, mempool.ErrDuplicate) {
 				// This is an actual exact duplicate tx, so
 				// return the specific duplicate tx error.
 				return nil, rpcDuplicateTxError("%v", err)
 			}
+			if s.cfg.SyncMgr.RecentlyConfirmedTxn(hash) {
```

Comment:
Looking at the code a bit more there and also confirming on simnet, it doesn't do what is intended as written in the PR.

It looks like there are two cases to consider:

1) The tx is mined and its outputs have **not** yet been spent.  In that case, the following path will be hit, resulting in an `ErrAlreadyExists`:

https://github.com/decred/dcrd/blob/f6fd7d3b0ff0f3bf2f3e00a41daae09e60805ade/internal/mempool/mempool.go#L1419-L1429

2) The tx is mined and all of its outputs have also already been spent.  In that case, it will end up as an orphan as previously described.

So, given that, I believe there are two changes needed to have the desired behavior:

* Look for `ErrAlreadyExists` as well in the check for the type of error:  e.g. `if errors.Is(rErr, mempool.ErrDuplicate) || errors.Is(rErr, mempool.ErrAlreadyExists) { /* existing code */ }
* Check if the transaction is not in the list of accepted transactions (I covered one case in the example code in the previous comment via checking len for zero, but it should also really check then entire list when it has entries too) and it's in the recently confirmed transactions filter as above.